File: Leetcode/167-Two-Sum-II---Input-array-is-sorted/TwoSumII.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    if (!numbers || numbers.length <2) {
        return [-1,-1]
    }
    var left=0
    var right=numbers.length-1
    while (left<right) {
        var s = numbers[left] + numbers[right]
        if (s == target) {
            return [left+1, right+1]
        } else if (s > target) {
            right--
        } else {
            left++
        }
    }
    return [-1,-1]
};

File: Leetcode/167-Two-Sum-II---Input-array-is-sorted/TwoSumII.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        if numbers == None or len(numbers) == 0:
            return [-1,-1]
        left = 0
        length = len(numbers)
        right = length-1
        while left < right:
            temp = numbers[left] + numbers[right]
            if temp == target:
                return [left+1, right+1]
            elif temp < target:
                left+=1
            else:
                right-=1
        return [-1, -1]
        
